### PR TITLE
Rename, tidy and add mpCompatible

### PR DESCRIPTION
**Pull request by tevans78**
Rename API features to be consistent
Add mpCompatible dependencies

#build

### CODE REVIEW

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.jwt-1.0.feature [hunk 1]
```diff
@@ -3,6 +3,7 @@ symbolicName=com.ibm.websphere.appserver.org.eclipse.microprofile.jwt-1.0
 WLP-DisableAllFeatures-OnConflict: false
 visibility=private
 singleton=true
+-features=io.openliberty.mpCompatible-0.0
 -bundles=com.ibm.websphere.org.eclipse.microprofile.jwt.1.0; location:="dev/api/stable/,lib/";type=jar; mavenCoordinates="org.eclipse.microprofile.jwt:microprofile-jwt-auth-api:1.0"
```
Comment (Azquelt) [inline dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.jwt-1.0.feature:7]: need to remove `type=jar` here, since you're now relying on this jar being a bundle

Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.jwt-1.0.feature]: that fixed it locally, thank you

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpJwt-1.2/com.ibm.websphere.appserver.mpJwt-1.2.feature [hunk 2]
```diff
@@ -15,9 +15,9 @@ Subsystem-Name: MicroProfile JSON Web Token 1.2
   com.ibm.websphere.appserver.jsonp-1.1, \
   com.ibm.websphere.appserver.httpcommons-1.0, \
   com.ibm.websphere.appserver.mpConfig-2.0, \
+  com.ibm.websphere.appserver.org.eclipse.microprofile.jwt-1.2, \
   io.openliberty.mpCompatible-4.0
 -bundles=com.ibm.ws.security.mp.jwt,\
-  io.openliberty.org.eclipse.microprofile.jwt.1.2; location:="dev/api/stable/,lib/"; mavenCoordinates="org.eclipse.microprofile.jwt:microprofile-jwt-auth-api:1.2-RC1",\
```
Comment (Emily-Jiang) [inline dev/com.ibm.websphere.appserver.features/visibility/public/mpJwt-1.2/com.ibm.websphere.appserver.mpJwt-1.2.feature:20]: Why did you delete this? Is this in the api feature?

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4.feature [hunk 3]
```diff
@@ -5,7 +5,8 @@ singleton=true
 -features=com.ibm.websphere.appserver.javax.cdi-2.0, \
  com.ibm.websphere.appserver.javax.annotation-1.3, \
  com.ibm.websphere.appserver.javax.jaxrs-2.1, \
- com.ibm.websphere.appserver.org.eclipse.microprofile.config-1.3; ibm.tolerates:=1.4
+ com.ibm.websphere.appserver.org.eclipse.microprofile.config-1.3; ibm.tolerates:=1.4,\
+ io.openliberty.mpCompatible-0.0
```
Comment (Emily-Jiang) [inline dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4.feature:9]: Why was the following -bundles not deleted, as you have done for MP Health? Any difference between Rest Client and MP Health?

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpJwt-1.2/com.ibm.websphere.appserver.mpJwt-1.2.feature (hunk 2 above)
Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/public/mpJwt-1.2/com.ibm.websphere.appserver.mpJwt-1.2.feature:20]: yes, that bundle is the API

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpJwt-1.2/com.ibm.websphere.appserver.mpJwt-1.2.feature [hunk 4]
```diff
@@ -15,9 +15,9 @@ Subsystem-Name: MicroProfile JSON Web Token 1.2
   com.ibm.websphere.appserver.jsonp-1.1, \
   com.ibm.websphere.appserver.httpcommons-1.0, \
   com.ibm.websphere.appserver.mpConfig-2.0, \
+  com.ibm.websphere.appserver.org.eclipse.microprofile.jwt-1.2, \
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/public/mpJwt-1.2/com.ibm.websphere.appserver.mpJwt-1.2.feature:18]: @Emily-Jiang replaced by this private API feature

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4.feature (hunk 3 above)
Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4.feature:9]: You're right, the MP Rest Client API should be a private feature too.

Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4.feature:9]: Since the PB for this PR was green, probably best to change the Rest Client API in a follow-up PR.

**Review by Azquelt — APPROVED** [review]

**Comment by tevans78** [comment]
#run-libby-bot

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4.feature (hunk 3 above)
Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.rest.client-1.4.feature:9]: This _is_ the private API feature
